backend/models/courseunit.mgmodel.ts

```diff
@@ -0,0 +1,105 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+type ElementSkeleton = {
+  id: string;
+  x: number;
+  y: number;
+  w: number;
+  h: number;
+  content: string;
+};
+
+const ElementSkeletonSchema: Schema = new Schema({
+  x: {
+    type: Number,
+    required: true,
+  },
+  y: {
+    type: Number,
+    required: true,
+  },
+  w: {
+    type: Number,
+    required: true,
+  },
+  h: {
+    type: Number,
+    required: true,
+  },
+  content: {
+    type: String,
+    required: true,
+  },
+});
+
+type Page = {
+  id: string;
+  title: string;
+  type: "Lesson" | "Activity";
```

Comment:
Could we move `"Lesson" | "Activity"` out into its own type?
e.g. `export type PageType = "Lesson" | "Activity";`

We could have this type in `backend/types/courseTypes.ts` (you'll need to do a `git pull origin main` to see this `backend/types/` folder, I just added it yesterday 😅)